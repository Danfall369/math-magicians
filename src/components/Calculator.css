body {
  background-color: black;
}

.calc-container {
  display: flex;
  justify-content: center;
  align-items: center;
}

.calc-wrapper {
  background-color: white;
  width: 40rem;
  transform: translateY(30px);
  border-radius: 10px;
}

.rows {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: repeat(4, 1fr);
}

.buttons-wrapper {
  height: 4em;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 24px;
  background-color: #e0e0e0;
  color: black;
  flex: 1;
  margin: 1px;
  border-radius: 10px;
}

.buttons-wrapper:nth-child(4n) {
  background-color: #f5913e;
}

.buttons-wrapper:nth-child(17) {
  grid-column: 1 / span 2;
}

.buttons-wrapper:nth-child(19) {
  background-color: #f5913e;
}

.input-wrapper {
  background-color: #858693;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: flex-end;
  padding: 16px;
  border-radius: 5px;
}

.result {
  flex: 1;
  font-size: 24px;
  color: white;
}
