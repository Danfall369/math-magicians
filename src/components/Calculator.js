import './Calculator.css';
import Buttons from './Buttons';
import Input from './input';

function Calculator() {
  return (
    <div className="calc-container">
      <div className="calc-wrapper">
        <Input />
        <div className="rows">
          <Buttons symbol="AC" />
          <Buttons symbol="+/-" />
          <Buttons symbol="%" />
          <Buttons symbol="รท" color="#f5913e" />
        </div>
        <div className="rows">
          <Buttons symbol="7" />
          <Buttons symbol="8" />
          <Buttons symbol="9" />
          <Buttons symbol="X" color="#f5913e" />
        </div>
        <div className="rows">
          <Buttons symbol="4" />
          <Buttons symbol="5" />
          <Buttons symbol="6" />
          <Buttons symbol="-" color="#f5913e" />
        </div>
        <div className="rows">
          <Buttons symbol="1" />
          <Buttons symbol="2" />
          <Buttons symbol="3" />
          <Buttons symbol="+" color="#f5913e" />
        </div>
        <div className="rows">
          <Buttons symbol="0" />
          <Buttons symbol=" " />
          <Buttons symbol="." />
          <Buttons symbol="=" color="#f5913e" />
        </div>
      </div>
    </div>
  );
}
export default Calculator;
