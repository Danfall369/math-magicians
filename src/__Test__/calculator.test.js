import { render } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Calculator', () => {
  test('renders correctly', () => {
    const { container } = render(<Calculator />);
    expect(container.querySelector('.calc-title')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
