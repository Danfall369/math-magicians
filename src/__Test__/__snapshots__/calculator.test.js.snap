// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `
<div>
  <h1
    class="calc-title"
  >
    Let's do some Math!
  </h1>
</div>
`;
